remote/src/main/scala/org/apache/pekko/remote/transport/netty/NettyTransport.scala

```diff
@@ -514,11 +507,7 @@ class NettyTransport(val settings: NettyTransportSettings, val system: ExtendedA
       address <- addressToSocketAddress(Address("", "", settings.BindHostname, bindPort))
     } yield {
       try {
-        val newServerChannel = inboundBootstrap match {
-          case b: ServerBootstrap         => b.bind(address)
-          case b: ConnectionlessBootstrap => b.bind(address)
-          case _                          => throw new IllegalStateException() // won't happen, compiler exhaustiveness check pleaser
-        }
```

Comment:
This new code changes the logic. How are you so sure that inboundBootstrap can never be a ConnectionlessBootstrap? I prefer the old code, to be frank.

Reply:
yes, but there will never has a ConnectionlessBootstrap , because udp support is completed removed.

Reply:
I think it was let there because of out of sight.

Reply:
I made this change - #639 

Would that be tidier?

Reply:
Both ok, i did not want to change the return type, this can rebase on yours